### PR TITLE
merge fetchAssetBalances & getBridgeBalances

### LINKED ISSUE #46: Merge `getBridgeBalances` with `fetchAssetBalances`
...in `SupportedWalletProvider.tsx`, they are doing same task, there should be also integration test for the new merge function too.

### LINKED ISSUE #47: Avoid manually calling `initApi` on all pages, instead only set the api instance when it's ready
```
File: /Users/ken/Workspace/CENNZnet/app-hub/pages/index.tsx
08: 	useEffect(() => {
09: 		if (!api?.isConnected) {
10: 			initApi();
11: 		}
12: 	}, [api, initApi]);
```

Will that be efficient enough to do a single `useEffect` in the CENNZApiProvider instead?

```
const [api, setApi] = useState<Api>(null);

useEffect(() => {
	const instance = new Api({ provider: endpoint });
	instance.isReady.then(() => {
		setApi(instance);
		window.onunload = async () => await instance.disconnect();
	});
}, [endpoint]);
```

### LINKED ISSUE #76: In Pool section, the app crashes when select ETH

### PR DESCRIPTION
**Pull request by aidan-starke**
## Description

- merge fetchAssetBalances & getBridgeBalances
- closes #46 
- remove initApi
- closes #47
- closes #76 - will fix at a later date

#### Changes

- implement new fetchAssetBalances function over whole app
- wrap fetchAssetBalances in updateBalances useCallback() to update balances after a transaction
- add token symbol to balance display in wallet
- format balances to show 4 decimal points
- hide 0 balance assets in wallet & on bridge
- move BalanceInfo to types
- restrict Pool & Swap tokens to CENNZ/CPAY

#### Fixes

- as Api is setup on render, balances no longer hang

#### Removes

- initApi Function
- ApiRx
- checkCENNZnetBalance function

#### Adds

- More assets for Nikau:  DAI & WETH
- SVG's for new assets
- test fetchAssetBalances

### CODE REVIEW

File: utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const formatBalance = (balance) =>
+	balance - Math.round(balance) !== 0 ? balance.toFixed(4) : balance;
```
Comment (ken-centrality) [inline utils/index.ts:2]: as per our Slack convo
- [ ] Add parameter type and return type so it's clear what's the function is doing
- [ ] Add logic to check for balance that is less than 4 dp, in that case output straight as `< 0.0001`
- [ ] Need some unit testing for this function to cover all cases scenario

I also think it's fine to output the number with .0000 at the end, so you don't need to check if the number is integer or not, so `balance - Math.round(balance) !== 0 ? balance.toFixed(4) : balance;` can just be as `balance.toFixed(4)`

**Review by ken-centrality — CHANGES_REQUESTED** [review]
Nice and clean cut overall, I like how we move a good chunk of logic out of React tree, looks much nicer!

File: components/shared/ConnectWalletButton.tsx [hunk 2]
```diff
@@ -139,9 +139,7 @@ const ConnectWalletButton: FC<ConnectWalletButtonProps> = ({
 						backgroundColor: color,
 					},
 				}}
-				onClick={() => {
-					onClick;
-				}}
+				onClick={() => onClick()}
```
[the remark this answers is not in the document]
Comment (aidan-starke) [reply components/shared/ConnectWalletButton.tsx:142]: that's what I thought, but it breaks 
Screen Shot 2022-02-23 at 6 45 35 PM

File: .env.production [hunk 3]
```diff
@@ -1,3 +1,3 @@
-NEXT_PUBLIC_SUPPORTED_ASSETS=16000,16001,17002
+NEXT_PUBLIC_SUPPORTED_ASSETS=16000,16001,17001,17002,17003
 NEXT_PUBLIC_API_URL="wss://nikau.centrality.me/public/ws"
```
[the remark this answers is not in the document]
Comment (aidan-starke) [reply .env.production:2]: yup, should be okay with new changes to fetch genericAsset balances

File: components/shared/ConnectWalletButton.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ken-centrality) [reply components/shared/ConnectWalletButton.tsx:142]: I checked, it's type issue, `onClick` of button demands `React.MouseEventHandler<HTMLButtonElement>` but what we pass to it is a global "Function" type which is too generic.

so if you change `onClick` type to `MouseEventHandler<HTMLButtonElement>` then it should work.

pinging @eupston as I've also seen this `onClick` somewhere on your PR too

[the remark this answers is not in the document]
Comment (eupston) [reply components/shared/ConnectWalletButton.tsx:142]: will fix in PR 59:
https://github.com/cennznet/app-hub/pull/59

**Review by eupston — APPROVED** [review]
lgtm

**Review by ken-centrality — APPROVED** [review]

**Pull request merged**